### ISSUE TITLE
Classification predictions

### ISSUE THREAD

**Issue by KrisNguyen135** [cornellius-gp/gpytorch#1561]
(Full Colab code [here](https://colab.research.google.com/drive/1n0hZCclcGisOfw0pmTWW7c1Jfy55uWkn?usp=sharing).) I'm trying to set up a model for a simple 3-class, 2-dimensional classification problem using a softmax likelihood like so:
```
class GPClassifier(ApproximateGP):
    def __init__(self, train_x, num_classes):
        variational_distribution = CholeskyVariationalDistribution(train_x.size(0))
        
        variational_strategy = UnwhitenedVariationalStrategy(
            self, train_x, variational_distribution, learn_inducing_locations=False
        )
        variational_strategy = MultitaskVariationalStrategy(
            variational_strategy, num_tasks=num_classes
        )
        
        super(GPClassifier, self).__init__(variational_strategy)
        self.mean_module = gpytorch.means.ConstantMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(
            gpytorch.kernels.RBFKernel()
        )
    
    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        latent_pred = gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
        return latent_pred
    
    
model = GPClassifier(X, num_groups)
likelihood = gpytorch.likelihoods.SoftmaxLikelihood(
    num_classes=num_groups, mixing_weights=False
)
mll = VariationalELBO(likelihood, model, y.numel())
```

and I'm having some trouble visualizing the model's predictions on a grid.

When I plot the latent means using `f_pred.loc` where
```
model.eval()
likelihood.eval()

with torch.no_grad():
    f_pred = model(X_test)
```
I got the same plot for all three latents, which makes me suspect I'm doing something wrong.

When I plot the `probs` and `logits` of the output from the likelihood, I got different things corresponding to the three latents, but they are not as smooth as I would like. I'm looking for something similar to what we get when we do regression on the labels like [this tutorial](https://docs.gpytorch.ai/en/stable/examples/01_Exact_GPs/GP_Regression_on_Classification_Labels.html).

Any pointer would be appreciated. Thanks in advance!

**Comment by gpleiss** [comment]
Hi @KrisNguyen135 would you mind opening up a [Discussion topic](https://github.com/cornellius-gp/gpytorch/discussions) rather than an issue? Thanks!

**Comment by KrisNguyen135** [comment]
Done; thanks for the comment. Just for my own information, do I only go to issues when I think I've found a bug or if I have a feature request?

**Comment by gpleiss** [comment]
@KrisNguyen135 much appreciated! we try to reserve issues for bugs and feature requests. That helps us prioritize work on our end 😄

**Comment by KrisNguyen135** [comment]
Thanks for the clarification @gpleiss
